File: app/src/main/java/remix/myplayer/misc/handler/OnHandleMessage.java
package remix.myplayer.misc.handler;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by Remix on 2017/11/16.
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface OnHandleMessage {

  int what() default 0;

  String action() default "";
}

File: app/src/main/java/remix/myplayer/misc/handler/OnHandleMessage.kt
package remix.myplayer.misc.handler

/**
 * Created by Remix on 2017/11/16.
 */
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class OnHandleMessage(val what: Int = 0, val action: String = "")